Diffing_Engine/Compute/_IDiffing.cs

```diff
@@ -52,62 +52,99 @@ public static Diff Diffing(IEnumerable<object> pastObjs, IEnumerable<object> fol
 
-                    return DiffRevisions(pastRev, follRev, diffConfigCopy);
+                        return DiffRevisions(pastRev, follRev, dc);
+                    }
                 }
+
+                if (diffingType == DiffingType.Revision)
+                    return DiffingError(diffingType);
             }
 
             IEnumerable<IBHoMObject> bHoMObjects_past = pastObjs.OfType<IBHoMObject>();
             IEnumerable<IBHoMObject> bHoMObjects_following = followingObjs.OfType<IBHoMObject>();
 
-            if (!string.IsNullOrWhiteSpace(customDataIdKey))
+            // Check if the BHoMObjects all have a RevisionFragment assigned.
+            // If so, we may attempt the Revision diffing.
+            if (bHoMObjects_past.AllHaveRevisionFragment() && bHoMObjects_following.AllHaveRevisionFragment())
+            {
+                BH.Engine.Reflection.Compute.RecordNote($"Calling the diffing method '{nameof(DiffRevisionObjects)}'.");
+                return DiffRevisionObjects(bHoMObjects_past, bHoMObjects_following, dc);
+            }
+
+            // If a customDataKey was specified, use the Id found under that key in customdata to perform the Diffing.
+            if (diffingType == DiffingType.Automatic || diffingType == DiffingType.CustomDataId)
             {
-                if (bHoMObjects_past.Count() == pastObjs.Count() && bHoMObjects_following.Count() == followingObjs.Count())
+                if (diffingType == DiffingType.CustomDataId && !string.IsNullOrWhiteSpace(dc.CustomDataKey))
+                    return DiffingError(diffingType);
+
+                if (!string.IsNullOrWhiteSpace(dc.CustomDataKey) && bHoMObjects_past.Count() == pastObjs.Count() && bHoMObjects_following.Count() == followingObjs.Count())
                 {
                     BH.Engine.Reflection.Compute.RecordNote($"Calling the diffing method '{nameof(DiffWithCustomId)}'.");
-                    return DiffWithCustomId(bHoMObjects_past, bHoMObjects_following, customDataIdKey, diffConfigCopy);
+                    return DiffWithCustomId(bHoMObjects_past, bHoMObjects_following, dc.CustomDataKey, dc);
                 }
                 else
                     BH.Engine.Reflection.Compute.RecordWarning($"To perform the diffing based on an Id stored in the Custom Data, the inputs must be collections of IBHoMObjects.");
             }
 
-            // Check if the BHoMObjects all have a hashfragment assigned.
-            // If so, we may attempt the Revision diffing.
-            if (bHoMObjects_past.AllHaveHashFragment() && bHoMObjects_following.AllHaveHashFragment())
-            {
-                BH.Engine.Reflection.Compute.RecordNote($"Calling the diffing method '{nameof(DiffRevisionObjects)}'.");
-                return DiffRevisionObjects(pastObjs, followingObjs, diffConfigCopy);
-            }
+            // Check if the bhomObjects have a persistentId assigned.
+            List<object> reminder_past, reminder_following;
```

Comment:
```suggestion
            List<object> reminder_past;
            List<object> reminder_following;
```

For consistency with BHoM practices.